### PR TITLE
add static gzip support

### PR DESCRIPTION
**Pull request by compermisos**
#### Pull Request (PR) description
Add config to use "gzip_static" config (https://nginx.org/en/docs/http/ngx_http_gzip_static_module.html)

#### This Pull Request (PR) fixes the following issues
This PR not close open issues

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -86,6 +86,7 @@
   $gzip_proxied                                              = 'off',
   $gzip_types                                                = undef,
   Enum['on', 'off'] $gzip_vary                               = 'off',
+  Enum['on', 'off'] $gzip_static                             = 'off',
```
Comment (ekohl) [inline manifests/init.pp]: The manual suggests you can also use `always` as a valid value

File: manifests/config.pp [hunk 2]
```diff
@@ -72,6 +72,7 @@
   $gzip_proxied                   = $nginx::gzip_proxied
   $gzip_types                     = $nginx::gzip_types
   $gzip_vary                      = $nginx::gzip_vary
+  $gzip_static                      = $nginx::gzip_static
```
Comment (ekohl) [inline manifests/config.pp]: Could you fix the alignment?

File: manifests/init.pp (hunk 1 above)
Comment (compermisos) [reply manifests/init.pp]: added always

File: manifests/config.pp (hunk 2 above)
Comment (compermisos) [reply manifests/config.pp]: correct identation

File: templates/conf.d/nginx.conf.erb [hunk 3]
```diff
@@ -110,6 +110,12 @@ http {
   etag <%= @etag %>;
 
 <% end -%>
+<% if @gzip_static == 'on' -%>
```
Comment (ekohl) [inline templates/conf.d/nginx.conf.erb]: The alternative is to use:
```erb
<% if @gzip_static != 'off' -%>
   gzip_static       <%= @gzip_static %>;
<% end -%>
```

Comment (alexjfisher) [reply templates/conf.d/nginx.conf.erb]: Or should it be `Optional` as the [ngx_http_gzip_static module](https://nginx.org/en/docs/http/ngx_http_gzip_static_module.html) might not be available?

> This module is not built by default, it should be enabled with the --with-http_gzip_static_module configuration parameter.

([`ngx_http_gzip_module`](http://nginx.org/en/docs/http/ngx_http_gzip_module.html) is built by default and provides the options `gzip_vary`, `gzip_types` and `gzip_proxied`)

Comment (compermisos) [reply templates/conf.d/nginx.conf.erb]: > Or should it be `Optional` as the [ngx_http_gzip_static module](https://nginx.org/en/docs/http/ngx_http_gzip_static_module.html) might not be available?
> 
> > This module is not built by default, it should be enabled with the --with-http_gzip_static_module configuration parameter.
> 
> ([`ngx_http_gzip_module`](http://nginx.org/en/docs/http/ngx_http_gzip_module.html) is built by default and provides the options `gzip_vary`, `gzip_types` and `gzip_proxied`)

The default value is off, the static module is different to normal gzip module, the normal module compress "on the fly", and static serve pre-compressed files

Comment (compermisos) [reply templates/conf.d/nginx.conf.erb]: > The alternative is to use:
> 
> ```
> <% if @gzip_static != 'off' -%>
>    gzip_static       <%= @gzip_static %>;
> <% end -%>
> ```

change made

**Comment by ekohl** [comment]
Have you had time to look at writing a test?

**Comment by dhoppe** [comment]
@compermisos It would be great, if you could provide some tests for the new parameter.

**Comment by compermisos** [comment]
you can guide me to make some tests?

**Comment by igalic** [comment]
you can find us in #voxpupuli on feenode or the outlet community slack.

in general, i would start by looking how the surrounding parameters are tested

**Comment by saz** [comment]
As far as I can see, this PR isn't required anymore due to the merge of #1386

**Pull request closed without merging**